app/controllers/admin/apps_controller.rb

```diff
@@ -13,8 +13,26 @@ def checkout
     @products = CheckoutProduct.where(:active => true).count
   end
 
-  def ideal
-    @transactions = IdealTransaction.order(created_at: :desc)
-                                    .paginate(page: params[:page], per_page: params[:limit] ||= 20)
+  def payments
+    ## This feels wack tbh somebody please come up with a better way.
+    @transactions = if params[:search].blank?
```

Comment:
can't you just build up your query per parameter?
so you get 1 if-statement per parameter